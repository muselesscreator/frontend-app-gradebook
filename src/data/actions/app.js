import { StrictDict } from 'utils';
import { modalFieldKeys, localFilterKeys } from 'data/constants/app';
import { createActionFactory } from './utils';

export const dataKey = 'app';
const createAction = createActionFactory(dataKey);

const closeModal = createAction('closeModal');
/**
 * setCourseId(courseId)
 * loads courseID into local app state
 * @param {string} courseId - course ID from app context
 */
const setCourseId = createAction('setCourseId');
/**
  * setModalStateFromTable({ userEntry, subsection })
  * sets modalState
  * */
const setModalStateFromTable = createAction('setModalStateFromTable');
/**
 * setSearchValue(searchValue)
 * sets searchValue in local state
 * @param {string} searchValue - search input text
 */
const setSearchValue = createAction('setSearchValue');

const filterReducer = (payload) => (obj, key) => (
  (payload[key] !== undefined) ? { ...obj, [key]: payload[key] } : obj
);

/**
 * setLocalFilter(filterObject)
 * sets a number of localFilterKey fields on localFilters
 * @param {object} filterObject - an object of filter values
 */
const setLocalFilter = createAction('setLocalFilter', (filterObject) => ({
  payload: Object.keys(localFilterKeys).reduce(filterReducer(filterObject), {}),
}));

/**
 * setModalState(modalState)
 * sets a number of modalFieldKey fields on modalState
 * @param {object} modalState - an object of modal state values
 */
const setModalState = createAction('setModalState', (modalState) => ({
  payload: Object.keys(modalFieldKeys).reduce(filterReducer(modalState), {}),
}));

export default StrictDict({
  closeModal,
  setCourseId,
  setModalState,
  setSearchValue,
  setLocalFilter,
  setModalStateFromTable,
});
