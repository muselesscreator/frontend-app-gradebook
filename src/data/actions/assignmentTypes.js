import { createAction } from '@reduxjs/toolkit';
import { StrictDict } from 'utils';

const fetching = {
  error: createAction('assignmentTypes/fetching/error'),
  started: createAction('assignmentTypes/fetching/started'),
  received: createAction('assignmentTypes/fetching/received'),
};
const gotGradesFrozen = createAction('assignmentTypes/gotGradesFrozen');

const actions = StrictDict({
  fetching,
  gotGradesFrozen,
});
export default actions;
