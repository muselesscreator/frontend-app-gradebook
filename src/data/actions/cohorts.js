import { createAction } from '@reduxjs/toolkit';


const startedFetching = createAction('cohorts/startedFetching');
const errorFetching = createAction('cohorts/errorFetching');
const received = createAction('cohorts/received');

export {
  errorFetching,
  received,
  startedFetching,
};
