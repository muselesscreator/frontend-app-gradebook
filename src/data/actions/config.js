import { createAction } from '@reduxjs/toolkit';
import { StrictDict } from 'utils';

const gotBulkManagementConfig = createAction('config/gotBulkManagement');

export default StrictDict({
  gotBulkManagementConfig,
});
