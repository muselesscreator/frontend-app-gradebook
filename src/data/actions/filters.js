import { createAction } from '@reduxjs/toolkit';
import initialFilters from '../constants/filters';

const initialize = createAction('filters/initialize', ({
  assignment = initialFilters.assignment,
  assignmentType = initialFilters.assignmentType,
  track = initialFilters.track,
  cohort = initialFilters.cohort,
  assignmentGradeMin = initialFilters.assignmentGradeMin,
  assignmentGradeMax = initialFilters.assignmentGradeMax,
  courseGradeMin = initialFilters.courseGradeMin,
  courseGradeMax = initialFilters.assignmentGradeMax,
  includeCourseRoleMembers = initialFilters.includeCourseRoleMembers,
}) => ({
  payload: {
    assignment: { id: assignment },
    assignmentType,
    track,
    cohort,
    assignmentGradeMin,
    assignmentGradeMax,
    courseGradeMin,
    courseGradeMax,
    includeCourseRoleMembers: Boolean(includeCourseRoleMembers),
  },
}));

const reset = createAction('filters/reset'); // paylaod
const update = {
  assignment: createAction('filters/update/assignment'),
  assignmentType: createAction('filters/update/assignmentType'),
  assignmentLimits: createAction('filters/update/assignmentLimits'),
  courseGrade: createAction('filters/update/courseGrade'),
  includeCourseRoleMembers: createAction('filters/update/includeCourseRoleMembers'),
};

export {
  initialize,
  reset,
  update,
};
