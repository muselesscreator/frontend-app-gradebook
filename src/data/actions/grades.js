import { createAction } from '@reduxjs/toolkit';

const csvUpload = {
  started: createAction('grades/csvUpload/started'),
  finished: createAction('grades/csvUpload/finished'),
  error: createAction('grades/csvUpload/error'),
};
const bulkHistory = {
  received: createAction('grades/bulkHistory/received'),
  error: createAction('grades/bulkHistory/error'),
};
const fetching = {
  started: createAction('grades/fetching/started'),
  finished: createAction('grades/fetching/finished'),
  error: createAction('grades/fetching/error'),
};
const overrideHistory = {
  errorFetching: createAction('grades/overrideHistory/errorFetching'),
  received: createAction(
    'grades/overrideHistory/received',
    (data) => ({
      payload: {
        overrideHistory: data.overrideHistory,
        currentEarnedAllOverride: data.currentEarnedAllOverride,
        currentPossibleAllOverride: data.currentPossibleAllOverride,
        currentEarnedGradedOverride: data.currentEarnedGradedOverride,
        currentPossibleGradedOverride: data.currentPossibleGradedOverride,
        originalGradeEarnedAll: data.originalGradeEarnedAll,
        originalGradePossibleAll: data.originalGradePossibleAll,
        originalGradeEarnedGraded: data.originalGradeEarnedGraded,
        originalGradePossibleGraded: data.originalGradePossibleGraded,
      },
    }),
  ),
};

const received = createAction(
  'grades/received',
  (data) => ({
    payload: {
      grades: data.grades,
      cohort: data.cohort,
      track: data.track,
      assignmentType: data.assignmentType,
      headings: data.headings,
      prev: data.prev,
      next: data.next,
      courseId: data.courseId,
      totalUsersCount: data.totalUsersCount,
      filteredUsersCount: data.filteredUsersCount,
    },
  }),
);

const update = {
  request: createAction('grades/update/request'),
  success: createAction('grades/update/success'),
  failure: createAction('grades/update/failure', (courseId, error) => ({
    payload: { courseId, error },
  })),
};

const uploadOverride = {
  success: createAction('grades/uploadOverride/success'),
  failure: createAction('grades/uploadOverride/failure', (courseId, error) => ({
    payload: { courseId, error },
  })),
};

// These actions for google analytics only. Doesn't change redux state.
const downloadReport = {
  bulkGrades: createAction('grades/downloadReport/bulkGrades'),
  intervention: createAction('grades/downloadReport/intervention'),
};

const toggleGradeFormat = createAction('grades/toggleGradeFormat');

const banner = {
  open: createAction('grades/banner/open'),
  close: createAction('grades/banner/close'),
};

const doneViewingAssignment = createAction('grades/doneViewingAssignment');

export {
  banner,
  bulkHistory,
  csvUpload,
  doneViewingAssignment,
  downloadReport,
  fetching,
  overrideHistory,
  received,
  toggleGradeFormat,
  update,
  uploadOverride,
};
