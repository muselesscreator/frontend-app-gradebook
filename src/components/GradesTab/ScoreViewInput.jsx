import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FormControl, FormGroup, FormLabel } from '@edx/paragon';

import actions from 'data/actions';

/**
 * <ScoreViewInput />
 * redux-connected select control for grade format (percent vs absolute)
 */
export const ScoreViewInput = ({ toggleFormat }) => (
  <FormGroup controlId="ScoreView">
    <FormLabel>Score View:</FormLabel>
    <FormControl
      as="select"
      value="percent"
      onChange={toggleFormat}
    >
      <option value="percent">Percent</option>
      <option value="absolute">Absolute</option>
    </FormControl>
  </FormGroup>
);
ScoreViewInput.propTypes = {
  toggleFormat: PropTypes.func.isRequired,
};

export const mapDispatchToProps = {
  toggleFormat: actions.grades.toggleGradeFormat,
};

export default connect(() => ({}), mapDispatchToProps)(ScoreViewInput);
