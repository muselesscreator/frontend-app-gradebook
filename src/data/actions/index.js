import * as assignmentTypes from './assignmentTypes';
import * as cohorts from './cohorts';
import * as config from './config';
import * as grades from './grades';
import * as roles from './roles';
import * as tracks from './tracks';

export {
  assignmentTypes,
  cohorts,
  config,
  grades,
  roles,
  tracks,
};
