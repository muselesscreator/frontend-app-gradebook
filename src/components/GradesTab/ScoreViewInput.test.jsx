import React from 'react';
import { shallow } from 'enzyme';

import actions from 'data/actions';

import { ScoreViewInput, mapDispatchToProps } from './ScoreViewInput';

jest.mock('@edx/paragon', () => ({
  FormControl: () => 'FormControl',
  FormGroup: () => 'FormGroup',
  FormLabel: () => 'FormLabel',
}));

jest.mock('data/actions', () => ({
  __esModule: true,
  default: {
    grades: { toggleGradeFormat: jest.fn() },
  },
}));

describe('ScoreViewInput', () => {
  describe('component', () => {
    let props;
    let el;
    beforeEach(() => {
      props = { toggleFormat: jest.fn() };
      el = shallow(<ScoreViewInput {...props} />);
    });
    const assertions = [
      'select box with percent and absolute options',
      'onClick from props.toggleFormat',
    ];
    test(`snapshot - ${assertions.join(' and ')}`, () => {
      expect(el).toMatchSnapshot();
    });
  });
  describe('mapDispatchToProps', () => {
    test('toggleFormat from actions.grades.toggleGradeFormat', () => {
      expect(mapDispatchToProps.toggleFormat).toEqual(actions.grades.toggleGradeFormat);
    });
  });
});
