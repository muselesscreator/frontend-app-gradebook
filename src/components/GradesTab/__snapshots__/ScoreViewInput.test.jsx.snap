// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreViewInput component snapshot - select box with percent and absolute options and onClick from props.toggleFormat 1`] = `
<FormGroup
  controlId="ScoreView"
>
  <FormLabel>
    Score View:
  </FormLabel>
  <FormControl
    as="select"
    onChange={[MockFunction]}
    value="percent"
  >
    <option
      value="percent"
    >
      Percent
    </option>
    <option
      value="absolute"
    >
      Absolute
    </option>
  </FormControl>
</FormGroup>
`;
